substrate/bin/utils/chain-spec-builder/src/lib.rs

```diff
@@ -279,18 +288,55 @@ pub struct VerifyCmd {
 	pub input_chain_spec: PathBuf,
 }
 
-/// Processes `CreateCmd` and returns JSON version of `ChainSpec`.
-pub fn generate_chain_spec_for_runtime(cmd: &CreateCmd) -> Result<String, String> {
-	let code = fs::read(cmd.runtime_wasm_path.as_path())
-		.map_err(|e| format!("wasm blob shall be readable {e}"))?;
+#[derive(Deserialize, Serialize, Clone)]
+pub struct ParachainExtension {
+	/// The relay chain of the Parachain.
+	pub relay_chain: String,
+	/// The id of the Parachain.
+	pub para_id: u32,
+}
 
-	let chain_type = &cmd.chain_type;
+/// Creates a new spec for a parachain.
+fn create_para_chain_spec(
+	cmd: &CreateCmd,
+	code: &[u8],
+	chain_type: ChainType,
+) -> sc_chain_spec::ChainSpecBuilder<ParachainExtension> {
+	let relay_chain = &cmd.relay_chain.clone().expect("No relay chain provided");
+	let para_id = *&cmd.para_id.expect("No Para ID provided");
 
-	let builder = GenericChainSpec::<()>::builder(&code[..], Default::default())
+	let mut properties = sc_chain_spec::Properties::new();
+	properties.insert("tokenSymbol".into(), "UNIT".into());
+	properties.insert("tokenDecimals".into(), 12.into());
+	properties.insert("ss58Format".into(), 42.into());
+
+	GenericChainSpec::<ParachainExtension>::builder(
```

Comment:
We don't really need extension and generic type parameter here. We could simply merge two json blobs using (json merge)[https://paritytech.github.io/polkadot-sdk/master/sc_chain_spec/json_patch/fn.merge.html] in case we generate for para.

But for now it is fine.

Reply:
So essentially create two chain specs (our "custom" para one then a generic one), then merge them?

Reply:
I've changed this to merging jsons.